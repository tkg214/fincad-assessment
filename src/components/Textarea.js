import React from 'react';
import PropTypes from 'prop-types';

const Textarea = props => {
  return (
    <fieldset className="form-group">
      <label data-test="label">
        {props.inputLabel}
      </label>
      <textarea
        className="form-control"
        id={props.inputId}
        value={props.inputValue}
        placeholder={props.inputPlaceholder}
        onChange={props.handleInputChange}
        data-test="textarea"
      />
    </fieldset>
  );
};

Textarea.propTypes = {
  inputLabel: PropTypes.string.isRequired,
  inputId: PropTypes.string.isRequired,
  inputValue: PropTypes.string.isRequired,
  inputPlaceholder: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired
};

export default Textarea;
