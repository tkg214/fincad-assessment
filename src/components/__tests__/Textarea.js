import React from 'react';
import { render, mount } from 'enzyme';

import Textarea from '../Textarea';

test('the textarea component renders with defaults', () => {
  const wrapper = renderInput();
  expect(wrapper).toMatchSnapshot();
});

test('the label renders correct props', () => {
  const inputLabel = 'Test Label';
  const wrapper = mountInput({ inputLabel });
  expect(wrapper.find(sel('label')).text()).toBe(inputLabel);
});

test('the textarea component invokes the handleInputChange prop when value changes', () => {
  const handleInputChange = jest.fn();
  const wrapper = mountInput({ handleInputChange });
  changeInput(wrapper);
  expect(handleInputChange).toHaveBeenCalledTimes(1);
});

function getInputProps(props = {}) {
  return {
    inputLabel: '',
    inputType: 'text',
    inputId: '',
    inputValue: '',
    inputPlaceholder: '',
    handleInputChange: () => {},
    ...props
  };
}

function renderInput(props = {}) {
  const inputProps = getInputProps(props);
  return render(<Textarea {...inputProps} />);
}

function mountInput(props = {}) {
  const inputProps = getInputProps(props);
  return mount(<Textarea {...inputProps} />);
}

function changeInput(wrapper) {
  wrapper
    .find(sel('textarea'))
    .simulate('change', { target: { value: 'test' } });
}

function sel(id) {
  return `[data-test="${id}"]`;
}
