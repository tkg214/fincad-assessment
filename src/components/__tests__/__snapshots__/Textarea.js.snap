// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the textarea component renders with defaults 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "form-group",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "data-test": "label",
            },
            "children": Array [],
            "name": "label",
            "next": Object {
              "attribs": Object {
                "class": "form-control",
                "data-test": "textarea",
                "id": "",
                "placeholder": "",
              },
              "children": Array [],
              "name": "textarea",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "class": "form-control",
              "data-test": "textarea",
              "id": "",
              "placeholder": "",
            },
            "children": Array [],
            "name": "textarea",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "data-test": "label",
              },
              "children": Array [],
              "name": "label",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
            },
            "type": "tag",
          },
        ],
        "name": "fieldset",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
